### PR TITLE
[IMP] hr_holidays: Improve business tests

### PR DESCRIPTION
**Pull request by jbm-odoo**
Description of the issue/feature this PR addresses:
The tests have been detroyed while merging half digested crap for the 12.0

Desired behavior after PR is merged:

new Unit tests to checks each business cases from "The Flow" into python tests.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_holidays/tests/test_accrual_requests.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        """
+        self.env.cr.execute("""
+                       UPDATE
+                       hr_employee
+                       SET create_date = '%s'
+                       WHERE id = %s
+                       """ % (newdate, id))
+
+    @mute_logger('odoo.models.unlink', 'odoo.addons.mail.models.mail_mail')
+    def test_accrual_interlapping(self):
```
Comment (RomainLibert) [inline addons/hr_holidays/tests/test_accrual_requests.py]: We should rename this to `test_overlapping_requests`

File: addons/hr_holidays/tests/test_accrual_requests.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        hol.sudo(self.user_hrmanager_id).write({
+            'date_from': (datetime.today() + relativedelta(days=4)),
+            'date_to': (datetime.today() + relativedelta(days=7)),
+            'number_of_days': 4,
+        })
+        with self.assertRaises(ValidationError):
+            hol.sudo(self.user_hrmanager_id).action_confirm()
+
+    @mute_logger('odoo.models.unlink', 'odoo.addons.mail.models.mail_mail')
+    def test_accrual_validity_time_valid(self):
```
Comment (RomainLibert) [inline addons/hr_holidays/tests/test_accrual_requests.py]: I suggest renaming this to `test_type_validity` and merging it with the next one

File: addons/hr_holidays/tests/test_holidays_flow.py [hunk 3]
```diff
@@ -118,31 +103,6 @@ def _check_holidays_status(holiday_status, ml, lt, rl, vrl):
         # Case2: limited type of leave request
```
Comment (RomainLibert) [inline addons/hr_holidays/tests/test_holidays_flow.py]: We could split this into two flows, using the comments, you'd have one testing in the case of an hr.leave.type which would not be limited and another with limited hr.leave.type

**Comment by RomainLibert** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**